### PR TITLE
azurerm_storage_share - minimum quota value correction

### LINKED ISSUE #4782: Failed to create Azure File Share when quota is defined
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azurerm_storage_share`

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "rg" {
  name     = "nancyc-rg-1"
  location = "westus2"
}

resource "azurerm_storage_account" "sa" {
  name                     = "nancyctest1234"
  resource_group_name      = azurerm_resource_group.rg.name
  location                 = azurerm_resource_group.rg.location
  account_kind             = "FileStorage"
  account_tier             = "Premium"
  account_replication_type = "LRS"
}

resource "azurerm_storage_share" "ss" {
  name                 = "bits"
  storage_account_name = azurerm_storage_account.sa.name
  quota                = 50
}
```

### Debug Output
[20191031.log](https://github.com/terraform-providers/terraform-provider-azurerm/files/3795196/20191031.log)

### Actual Behavior
Error:

2019/10/31 10:57:28 [DEBUG] azurerm_storage_share.ss: apply errored, but we're indicating that via the Error pointer rather than returning it: Error creating Share "bits" (Account "nancyctest1234" / Resource Group "nancyc-rg-1"): shares.Client#Create: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: error response cannot be parsed: "\ufeff<?xml version=\"1.0\" encoding=\"utf-8\"?><Error><Code>InvalidHeaderValue</Code><Message>The value for one of the HTTP headers is not in the correct format.\nRequestId:942190c5-401a-003c-2c14-9033e1000000\nTime:2019-10-31T17:57:28.0079937Z</Message><HeaderName>x-ms-share-quota</HeaderName><HeaderValue>50</HeaderValue></Error>" error: invalid character 'ï' looking for beginning of value

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

### PR DESCRIPTION
**Pull request by DanielMabbett**
This will fix the issue: 
https://github.com/terraform-providers/terraform-provider-azurerm/issues/4782

Changes: 
+ Change the quota minimum validation value to 100 
+ Change the test quota value to 100. 

Fixes #4782

### CODE REVIEW

**Comment by DanielMabbett** [comment]
image

**Comment by DanielMabbett** [comment]
It appears this may need some more work. Quota may change depending on Premium or Standard

**Comment by DanielMabbett** [comment]
I have taken out the file changes due to the provider not currently able to change validation depending on the storage account kind. 

Tom says that this error message will become clearer in the next release: 
> hi @yupwei68
> 
> Thanks for opening this issue.
> 
> This UTF8-parsing error is coming from a bug in Azure/Go-AutoRest (where errors were parsed as Json always, rather than as XML if XML is in the response) which has since been fixed and has shipped in version 0.6.0 of this library - which will ship in v1.37 of the Terraform Provider for Azure.
> 
> As such I'm going to close this issue for the moment - but the error message should become clearer in v1.37 of Terraform's Azure Provider. It's likely this is to do with the minimum quota size changing depending on the Storage Account Type being used (e.g. Standard vs Premium) - as is detailed in [terraform-providers/terraform-provider-azurerm#4782](https://github.com/terraform-providers/terraform-provider-azurerm/issues/4782)
> 
> Thanks!
https://github.com/tombuildsstuff/giovanni/issues/20

So, for now, a documentation update describing the limits on the argument should do.

File: website/docs/r/storage_share.html.markdown [hunk 1]
```diff
@@ -44,7 +44,7 @@ The following arguments are supported:
 
 * `acl` - (Optional) One or more `acl` blocks as defined below.
 
-* `quota` - (Optional) The maximum size of the share, in gigabytes. Must be greater than 0, and less than or equal to 5 TB (5120 GB) for Standard storage accounts or 100 TB (102400 GB) for Premium storage accounts. Default is 5120.
+* `quota` - (Optional) The maximum size of the share, in gigabytes. For Standard storage accounts, this must be greater than 0 and less than 5 TB (5120 GB). For Premium FileStorage storage accounts, this must be greater than 100 GB and less than 100 TB (102400 GB). Default is 5120.
```
Comment (katbyte) [inline website/docs/r/storage_share.html.markdown]: Could we word this a big different:
```suggestion
* `quota` - (Optional) The maximum size of the share, in gigabytes. For Standard storage accounts, this must be greater than 0 and less than 5120 GB (5 TB). For Premium FileStorage storage accounts, this must be greater than 100 GB and less than 102400 GB (100 TB). Default is 5120.
```

**Comment by DanielMabbett** [comment]
Sure thing. Happy to help.

**Pull request merged**